### PR TITLE
feat!: add memo to IBC transfers of ICS rewards

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: #1634

TODO:

- [x] add e2e tests
- [x] add info about migration needed on the consumer (new param needs to be initialized)
- [x] add changelog entries
- [ ] backport to v4.5.x so it can be used by consumers with SDK 0.47

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if the change is [state-machine breaking](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#breaking-changes)
* [ ] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/00-intro.md)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` the type prefix if the change is state-machine breaking
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

**Review by insumity — APPROVED** [review]
The draft looks good to me.
Did you talk with @MSalopek about his solution that doesn't involve consumer changes?

File: .changelog/unreleased/features/2290-add-memo-to-ICS-rewards.md.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- `[x/consumer]` Populate the memo on the IBC transfer packets used to send ICS rewards.
```
Comment (insumity) [inline .changelog/unreleased/features/2290-add-memo-to-ICS-rewards.md.md:1]: nit: file name contains twice `.md`, hence `.md.md`

**Review by insumity — APPROVED** [review]
Amazing work!

**Review by MSalopek — APPROVED** [review]
Admin approval

**Pull request merged**